### ISSUE TITLE
Warn if deploy config doesn't match source config

### ISSUE THREAD

**Issue by srid** [obsidiansystems/obelisk#179]
If we add a new config file in source, and try to deploy it later `ob deploy` won't complain (it doesn't have the added config file in the deploy repo).

**Comment by eskimor** [comment]
I think there should be a way to specify that certain configs are required for operation and the deployment will just fail if they are not present. This could be accompanied with a `getRequired` function that is total.

Something along those lines is quite important, as configs have to be given individually for each deployment - so even if you test on a test server you could miss some config on production and screw it up.

Something like this would also be nice for usability, as `ob deploy` could spit out a message like

```
ERROR: The following required config files are missing:
 - config/common/very-important-stuff
 - config/backend/even-more-important stuff
```

Ideally we would not only test that a config is present, but also that it is valid. But this would be much more involved, as checking that a config is valid very easily requires the application to be run. (E.g. oauth configuration)

A quick hack to get some compile time guarantees about the presence of config files, would be to try to read them with TH and spit out an error at compile time if that fails.

**Comment by ryantrinkle** [comment]
Yes, I'd like us to eventually parse and possibly even validate the configs
before deploying them; however, in the mean time, just checking that the
set of files matches sounds like a good intermediate step.  I don't think
we should bother with "optional" files, though; all files should be
required, and if the user wants to have a "Nothing" value they can encode
that into the contents of the file instead.
